data_model/derive/src/event_set.rs

```diff
@@ -0,0 +1,395 @@
+
+                        fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+                            formatter.write_str("a sequence of strings")
+                        }
+
+                        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+                        where
+                            A: serde::de::SeqAccess<'de>,
+                        {
+                            let mut result = #set_ident::empty();
+
+                            struct SingleEvent(#set_ident);
+                            impl<'de> serde::Deserialize<'de> for SingleEvent {
+                                fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+                                where
+                                    D: serde::Deserializer<'de>,
+                                {
+                                    struct Visitor;
+
+                                    impl<'de> serde::de::Visitor<'de> for Visitor {
+                                        type Value = SingleEvent;
+
+                                        fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+                                            formatter.write_str("a string")
+                                        }
+
+                                        fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+                                        where
+                                            E: serde::de::Error,
+                                        {
+                                            let event = match v {
+                                                #(
+                                                    #flag_names => #flag_idents,
+                                                )*
+                                                _ => return Err(serde::de::Error::custom(format!("unknown event variant `{}`, expected one of {}", v, #flag_names_str))),
+                                            };
+
+                                            Ok(SingleEvent(event))
+                                        }
+                                    }
+
+                                    deserializer.deserialize_str(Visitor)
+                                }
+                            }
+                            while let Some(SingleEvent(event)) = seq.next_element::<SingleEvent>()? {
+                                result |= event;
+                            }
+
+                            Ok(result)
+                        }
+                    }
+
+                    deserializer.deserialize_seq(Visitor)
+                }
+            }
+        })
+    }
+}
+
+pub fn impl_event_set_derive(emitter: &mut Emitter, input: &syn::DeriveInput) -> TokenStream {
+    let Some(enum_) = emitter.handle(EventSetEnum::from_derive_input(input)) else {
+        return quote! {};
+    };
+
+    quote! {
+        #enum_
```

Comment:
does it get wrapped with the `model ` macro?

Reply:
I doesn't, but should it? As far as I understand, `model!` macro is needed for controlling access to internal fields and the generated `*EventSet` structures don't have any, so it would work as a no-op.

I guess I can additionally wrap it (and that wouldn't even be a problem for testing, as the tests do not access the internal field), but it would not change the exposed API in any meaningful way